test/config/featureSwitches/FeatureSwitchSpec.scala

```diff
@@ -147,4 +148,84 @@ class FeatureSwitchSpec extends SpecBase {
+    }
+
+    s"remove an existing date when the parameter is $None" in new Setup {
+      featureSwitching.setEstimatedLPP1FilterEndDate(Some(dateTimeNow))
+      (sys.props get featureSwitching.ESTIMATED_LPP1_FILTER_END_DATE) shouldBe Some(dateTimeNow.toString)
+      featureSwitching.setEstimatedLPP1FilterEndDate(None)
+      (sys.props get featureSwitching.ESTIMATED_LPP1_FILTER_END_DATE) shouldBe None
+    }
+  }
+
+  "FeatureSwitching getEstimatedLPP1FilterEndDate" should {
+    lazy val dateTimeNowMinus1Day: LocalDateTime = LocalDateTime.parse(LocalDateTime.now().format(DateHelper.dateTimeFormatter), DateHelper.dateTimeFormatter).minusDays(1)
+    s"get the date when it exists in system properties" in new Setup {
+      featureSwitching.setEstimatedLPP1FilterEndDate(Some(dateTimeNowMinus1Day))
+      (sys.props get featureSwitching.ESTIMATED_LPP1_FILTER_END_DATE) shouldBe Some(dateTimeNowMinus1Day.toString)
+      featureSwitching.getEstimatedLPP1FilterEndDate shouldBe dateTimeNowMinus1Day
+    }
+
+    s"get the date from config when the key value exists and is non-empty" in new Setup {
+      (sys.props get featureSwitching.ESTIMATED_LPP1_FILTER_END_DATE) shouldBe None
+      when(mockConfig.getOptional[String](any())(any()))
+        .thenReturn(Some(dateTimeNowMinus1Day.toString))
+      featureSwitching.getEstimatedLPP1FilterEndDate shouldBe dateTimeNowMinus1Day
+    }
+
+    s"get the date from the system when it does not exist in properties nor in config (value empty)" in new Setup {
+      (sys.props get featureSwitching.ESTIMATED_LPP1_FILTER_END_DATE) shouldBe None
+      when(mockConfig.getOptional[String](any())(any()))
+        .thenReturn(Some(""))
+      featureSwitching.getEstimatedLPP1FilterEndDate.toLocalDate shouldBe LocalDate.now().minusDays(1) //Set to LocalDate to stop flaky tests
+    }
+
+    s"get the date from the system when it does not exist in properties nor in config (kv not present)" in new Setup {
+      (sys.props get featureSwitching.ESTIMATED_LPP1_FILTER_END_DATE) shouldBe None
+      when(mockConfig.getOptional[String](any())(any()))
+        .thenReturn(None)
+      featureSwitching.getEstimatedLPP1FilterEndDate.toLocalDate shouldBe LocalDate.now().minusDays(1) //Set to LocalDate to stop flaky tests
+    }
+  }
+
+  "FeatureSwitching withinLPP1FilterWindow" should {
+
+    "return true" when{
+      "now is before set date" in new Setup {
+        featureSwitching.setEstimatedLPP1FilterEndDate(Some(LocalDateTime.now().plusDays(1)))
+        featureSwitching.withinLPP1FilterWindow shouldBe true
+      }
+    }
+
+    "return false" when{
+      "now is after set date" in new Setup {
+        featureSwitching.setEstimatedLPP1FilterEndDate(Some(LocalDateTime.now().minusDays(1)))
+        featureSwitching.withinLPP1FilterWindow shouldBe false
+      }
+
+      "an end date is mot set (Default to Off)" in new Setup {
```

Comment:
```suggestion
      "an end date is not set (Default to Off)" in new Setup {
```
Not sure about (Default to Off)? What does that mean?